dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -139,20 +140,34 @@ def _merge_words(self, other, merged_profile):
         :type merged_profile: TextProfiler
         :return:
         """
-        if not self._is_case_sensitive:
-            merged_profile.word_count = self.word_count.copy()
-            additive_words = other.word_count
+        if self._is_case_sensitive == other._is_case_sensitive:
+            merged_profile.word_count = self.word_count + other.word_count
         else:
-            merged_profile.word_count = other.word_count.copy()
-            additive_words = self.word_count
-
-        for word in additive_words:
-            word_lower = word.lower()
-            if word_lower not in self._stop_words:
-                if self._is_case_sensitive:
-                    merged_profile.word_count[word] += additive_words[word]
-                else:
-                    merged_profile.word_count[word_lower] += additive_words[word]
+            if not self._is_case_sensitive:
+                non_case_sensitive_word_count = self.word_count
+                case_sensitive_word_count = other.word_count
+            else:
+                non_case_sensitive_word_count = other.word_count
+                case_sensitive_word_count = self.word_count
+
+            additive_word_count = Counter()
+            for k, v in case_sensitive_word_count.items():
+                additive_word_count.update({k.lower(): v})
+            merged_profile.word_count = \
+                non_case_sensitive_word_count + additive_word_count
```

Comment:
`merge counts`
`if either is not case sensitive (not both case sensitive):
        update with lowers'

Reply:
if we want to make it this way due to the readability, I can do it. But it'll be less efficient than the current implementation as we have to go through two counters to lowercase later.

Reply:
changed